### PR TITLE
Model-View-Wizard!

### PR DESCRIPTION
**Pull request by mistergone**
This PR moves the code into a MV* framework I'm calling "Model-View-Wizard"! Wizards have got it going on.

## Additions
- Several files (which just have code that used to be in two big files)

## Changes
- Moved code into a bunch of smaller files.
- Unit tests changes to use new files

## Testing
- Pull down, run `frontendbuild.sh`, and make sure everything works!

## Review
- @marteki and @niqjohnson and @higs4281 

## Todos
- The graph view is still big and messy, so refactoring it on my To Do List.

## Checklist
* [x] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices 
* [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

## Special Thanks
Thanks to @marteki and @niqjohnson for their patience while I rework all this stuff. Thanks to @ascott1 for help setting up `college-costs`, which became a template for this project!

## Animated GIF
pretty-pretty-good

### CODE REVIEW

**Comment by mistergone** [comment]
@higs4281 - I think you knew this, but Travis hates your numbers again. :smile:

**Comment by mistergone** [comment]
Fix for test pushed!

**Comment by higs4281** [comment]
:+1: At 1/1/1949 $40,000 the graph shows a stray $0 over in the blank age 66 column.
I think this will be cleared up with the reborn-on-1st PR, so it shouldn't block this.

**Comment by niqjohnson** [comment]
Woo! Looking good. I saw a couple of JS errors while poking around.

First, If you try to resize the window before you've hit the "get your estimates" button, you'll get a console error and the graph won't render correctly after you hit that button:

```
graph-view.js:270 Uncaught TypeError: Cannot read property 'attr' of undefined
```

Second, when you select an age in the step 3 dropdown, you'll get a console error and the page won't auto-scroll to the top of the revealed content:

```
graph-view.js:88 Uncaught ReferenceError: isElementInView is not defined
```

**Comment by marteki** [comment]
Working as far as I can see on standalone, with no stray $0s appears. (My UnityBox is apparently borked, so I have to get it back up and running before I can look at it there.)

:+1:

**Comment by marteki** [comment]
Note: I just confirmed the JS errors @niqjohnson found on standalone. :worried:

**Comment by mistergone** [comment]
This is why I love @niqjohnson - I'm going to work on those JS errors real fast (although at least one will be irrelevant with my next PR)! I'll look into Bill's error as well!

**Comment by mistergone** [comment]
JS errors are fixed!

**Comment by mistergone** [comment]
@higs4281 - In the case of 1/1/1949, the API is giving back a `data.['full retirement age']` of `"66"`. The interface thinks it needs to put a number over the FRA, but the benefits data for 66 is 0. So it puts $0 over age 66. Should I be writing an exception into the front-end for this case, or is the API out of date or is it something else entirely? :smile:

**Comment by mistergone** [comment]
There were some changes to `master` that I must have missed! I pulled them in, but I still get `"66"` as FRA for 1/1/1949

**Comment by marteki** [comment]
Might PR https://github.com/cfpb/retirement/pull/140 resolve the slightly-off FRA problem? I can't replicate, so need to ask: is the rogue $0 only happening for folks born on the 1st?

**Comment by mistergone** [comment]
Hrm. I see it for anyone whose age > FRA. So maybe I should just have that box hidden when someone's `currentAge > FRA`? It might just be a new bug I introduced that has nothing to do with the API value being returned. In fact, I bet that's it.

**Comment by higs4281** [comment]
Current age is 67, past the FRA of 66. We should show no numbers for years before current age.

This didn't happen before. Some rule somewhere was blanking out anything before current age.

**Comment by marteki** [comment]
A bugfix that went through late last year, in regards to not allowing the indicator to be moved before the current age: https://github.com/cfpb/retirement/pull/125

And yes, we shouldn't be showing or allowing people to select anything in years earlier than their current age, across the board.

**Comment by mistergone** [comment]
Yep, I got confused about the definition of FRA (my brain said, "Well, FRA changes when you get older than it, right?"). :smile: Working on this now!

**Comment by mistergone** [comment]
The fix for erroneously-appearing full benefits text when the user is older than FRA is in! This PR should be good now. Please give your thumbs.

**Comment by mistergone** [comment]
I also pushed a fix for users past FRA who would see an increase of __Infinity%__, which is not technically correct.

File: src/js/views/graph-view.js [hunk 1]
```diff
@@ -0,0 +1,626 @@
...
+      $( '.graph-content .content-container.max-retirement' ).show();
+    }
+    else if ( this.selectedAge < SSData.fullAge ) {
+      $( '.graph-content .content-container.early-retirement' ).show();
+    }
+    else {
+      $( '.graph-content .content-container.full-retirement' ).show();
+    }
+
+    console.log( this.selectedAge, SSData.fullAge, SSData );
```
Comment (marteki) [inline src/js/views/graph-view.js]: Hey, Chuck! You and I always do this... :smile:

**Comment by marteki** [comment]
Aside from the `console.log`, everything else looks ready to ship! :+1:

**Comment by mistergone** [comment]
What console.log()? :wink:

**Comment by niqjohnson** [comment]
:+1: 

Tested in standalone on Chrome and in UnityBox on iPhone. Everything seemed to be working for me. Nice!

**Comment by mistergone** [comment]
Requisite thumbs reached - __merging!__

**Pull request merged**